src/notebook/notebook/nbformat.ts

```diff
@@ -88,13 +88,61 @@ namespace nbformat {
    */
   export
   interface MimeBundle extends JSONObject {
-    [key: string]: multilineString;
-    'application/json'?: any;
+    [key: string]: multilineString | JSONObject;
   }
   /* tslint:enable */
 
 
   /**
+   * Validate a mime type/value pair.
+   *
+   * @param type - The mimetype name.
+   *
+   * @param value - The value associated with the type.
+   *
+   * @returns Whether the type/value pair are valid.
+   */
+  export
+  function validateMimeValue(type: string, value: multilineString | JSONObject): boolean {
+    // Check if "application/json" or "application/foo+json"
+    const jsonTest = /^application\/(.*?)+\+json$/;
+    const isJSONType = type === 'application/json' || jsonTest.test(type);
```

Comment:
It's tempting to make all of our logic (and runtime) simpler by declaring nbformat to accept `.*json` as raw JSON object.